packages/loader/container-loader/src/container.ts

```diff
@@ -724,10 +714,7 @@ export class Container
 	/**
 	 * @internal
 	 */
-	constructor(
-		private readonly createProps: IContainerCreateProps,
-		loadProps?: IContainerLoadProps,
-	) {
+	constructor(createProps: IContainerCreateProps, pendingLocalState?: IPendingContainerState) {
```

Comment:
i'd prefer to keep load props here. it much easier to modify an object rather than modify a params list, and i think create vs load is a good distinction.

On thing is loses is that resolved url is required on load but not create, with this change it never required at the constructor layer

Reply:
Really I think it should be removed entirely if we can find a nice way - I believe originally IContainerCreateProps === constructor param, and IContainerLoadProps === container.load() param.  At some point this division got broken for pendingLocalState I guess.  As a compromise, I'll do a Pick here (same reasoning as (2) in the PR description - force a conversation if we think some member is required both at create and load time).